unstable test TestDBStmtCount

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

```
cd executor
go test -v -tags intest -run  TestDBStmtCount
```

### 2. What did you expect to see? (Required)

No error

### 3. What did you see instead (Required)

```
--- FAIL: TestDBStmtCount (0.67s)
    server_test.go:2160:
                Error Trace:    /home/genius/project/src/[github.com/pingcap/tidb/server/server_test.go:2160](http://github.com/pingcap/tidb/server/server_test.go:2160)
                                                        /home/genius/project/src/[github.com/pingcap/tidb/server/server_test.go:154](http://github.com/pingcap/tidb/server/server_test.go:154)
                                                        /home/genius/project/src/[github.com/pingcap/tidb/server/server_test.go:2123](http://github.com/pingcap/tidb/server/server_test.go:2123)
                                                        /home/genius/project/src/[github.com/pingcap/tidb/server/tidb_serial_test.go:1](http://github.com/pingcap/tidb/server/tidb_serial_test.go:1)
19
                Error:          Not equal:
                                expected: 3
                                actual  : 2
                Test:           TestDBStmtCount
FAIL
exit status 1
```
FAIL    [github.com/pingcap/tidb/server](http://github.com/pingcap/tidb/server)  0.710s

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

master e123433e2abda14e3540bc071d414f0a92e77725

--- a/server/server_test.go
+++ b/server/server_test.go
@@ -2120,6 +2120,11 @@ func (cli *testServerClient) runTestStmtCount(t *testing.T) {
 }
 
 func (cli *testServerClient) runTestDBStmtCount(t *testing.T) {
+	// When run this test solely, the test can be stable.
+	// But if other tests run during this, the result can be unstable.
+	// Because the test collect some metrics before / after the test, and the metrics data are polluted.
+	t.Skip("unstable test")
+
 	cli.runTestsOnNewDB(t, nil, "DBStatementCount", func(dbt *testkit.DBTestKit) {
 		originStmtCnt := getDBStmtCnt(string(cli.getMetrics(t)), "DBStatementCount")
 

